### PR TITLE
Make validation of unique node outputs much faster

### PR DESCRIPTION
**Pull request by deepyaman**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

The logic in `Node._validate_unique_outputs()` is convoluted--which led me to investigate--and slow.

## Development notes

I used a much simpler approach to achieve the same thing, very similar to https://stackoverflow.com/a/46554884/1093967. Some simple validation follows.

With duplicates:
```python
>>> timeit.timeit('set((Counter(l) - Counter(set(l))).keys())',"from collections import Counter;l = ['a', 'a', 'b', 'c', 'a', 'b']")
2.121045750012854
>>> timeit.timeit('Counter(l) - Counter(set(l))',"from collections import Counter;l = ['a', 'a', 'b', 'c', 'a', 'b']")  # Just in case there's any doubt about what the slow part is
2.0280554170021787
>>> timeit.timeit('c = Counter(l);{k for k in c if c[k] > 1}',"from collections import Counter;l = ['a', 'a', 'b', 'c', 'a', 'b']")
0.793446000025142
```

No duplicates:
```python
>>> timeit.timeit('set((Counter(l) - Counter(set(l))).keys())',"from collections import Counter;l = ['a', 'b', 'c']")
1.8756820000126027
>>> timeit.timeit('Counter(l) - Counter(set(l))',"from collections import Counter;l = ['a', 'b', 'c']")
1.7998937499942258
>>> timeit.timeit('c = Counter(l);{k for k in c if c[k] > 1}',"from collections import Counter;l = ['a', 'b', 'c']")
0.7210281250008848
```

Single repeated element:
```python
>>> timeit.timeit('set((Counter(l) - Counter(set(l))).keys())',"from collections import Counter;l = ['a'] * 1000")
28.893976499995915
>>> timeit.timeit('Counter(l) - Counter(set(l))',"from collections import Counter;l = ['a'] * 1000")
28.780192749982234
>>> timeit.timeit('c = Counter(l);{k for k in c if c[k] > 1}',"from collections import Counter;l = ['a'] * 1000")
20.062867875007214
```

It's not even close.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: kedro/pipeline/node.py [hunk 1]
```diff
@@ -480,11 +480,12 @@ def _validate_inputs(self, func, inputs):
                 ) from exc
 
     def _validate_unique_outputs(self):
-        diff = Counter(self.outputs) - Counter(set(self.outputs))
+        c = Counter(self.outputs)
+        diff = {k for k in c if c[k] > 1}
```
Comment (deepyaman) [inline kedro/pipeline/node.py:484]:
```suggestion
        cnt = Counter(self.outputs)
        diff = {k for k in cnt if cnt[k] > 1}
```

**Comment by datajoely** [comment]
Cool!

**Review by astrojuanlu — APPROVED** [review]
Nice and short 👍🏽

**Review by merelcht — APPROVED** [review]
Thanks for improving this @deepyaman ⭐!

**Pull request merged**